### PR TITLE
bug 620 App crashes on dev - HeatMap

### PR DESCRIPTION
**Pull request by danielsh28**

### CODE REVIEW

File: src/components/molecules/HeatMap.tsx [hunk 1]
```diff
@@ -32,11 +32,13 @@ const HeatMap: FC<IProps> = ({ data, center, sizeOptions }) => {
 
   const mapRef = React.createRef<any>();
   useEffect(() => {
-    const map = mapRef.current.leafletElement;
-    setTimeout(() => {
-      map.invalidateSize();
-    });
-  }, [sizeOptions, mapRef]);
+    if(mapRef.current) {
```
Comment (yuvalbl) [inline src/components/molecules/HeatMap.tsx:35]:
```
if(mapRef.current?.leafletElement) {
      const {leafletElement} = mapRef.current.leafletElement;
      setTimeout(leafletElement.invalidateSize);
}
```

**Comment by danielsh28** [comment]
another issue of this bug is  described here[https://github.com/OpenGov/react-leaflet-heatmap-layer/issues/49](url)

File: src/components/molecules/HeatMap.tsx [hunk 2]
```diff
@@ -32,11 +32,13 @@ const HeatMap: FC<IProps> = ({ data, center, sizeOptions }) => {
...
+      const map = mapRef.current.leafletElement;
+      setTimeout(() => {
```
Comment (yuvalbl) [inline src/components/molecules/HeatMap.tsx:37]: add a comment here with reference to https://github.com/OpenGov/react-leaflet-heatmap-layer/issues/49

**Comment by danielsh28** [comment]
explanation to this bug can be in [here](https://github.com/Leaflet/Leaflet/issues/6981). 
As mentioned in the link, this issue can be resolved by updating the leaflet dependency from 1.6 to 1.7. I assume that  an update is needed to  all the  packages that use leaflet as a dependency

File: src/components/molecules/HeatMap.tsx (hunk 2 above)
Comment (eidan66) [reply src/components/molecules/HeatMap.tsx:37]: @yuvalbl  [this site](https://github.com/OpenGov/react-leaflet-heatmap-layer/issues/49) show 404 error.
maybe they delete the issues..

**Comment by yuvalbl** [comment]
should be solved by #655

**Pull request closed without merging**